pyomo/dae/tests/test_flatten.py

```diff
@@ -208,7 +292,997 @@ def c_rule(m, t):
+                assert len(comps) == len(ref_data)
+                for comp in comps:
+                    self.assertIn(self._hashRef(comp), ref_data)
+            else:
+                raise RuntimeError()
+
+    def test_flatten_m1_along_time(self):
+        m = self._model_1()
+        
+        sets = ComponentSet((m.time,))
+        sets_list, comps_list = flatten_components_along_sets(m, sets, Var)
+
+        S = m.space
+        SS = m.space*m.space
+        SC = m.space*m.comp
+        SSC = m.space*m.space*m.comp
+
+        assert len(sets_list) == 3
+        for sets, comps in zip(sets_list, comps_list):
+            if len(sets) == 1 and sets[0] is UnindexedComponent_set:
+                ref_data = {
+                        self._hashRef(Reference(m.v0)),
+                        }
+                assert len(comps) == len(ref_data)
+                for comp in comps:
+                    self.assertIn(self._hashRef(comp), ref_data)
+            elif len(sets) == 1 and sets[0] is m.time:
+                ref_data = {
+                    # Components indexed only by time;                        
+                    self._hashRef(Reference(m.v1)),
+                    self._hashRef(Reference(m.b.b1[:].v0)),
+                }
+                # Components indexed by time and some other set(s)
+                ref_data.update(self._hashRef(Reference(m.v2[:,x])) for x in S)
+                ref_data.update(self._hashRef(Reference(m.v3[:,x,j])) for x, j in SC)
+                ref_data.update(self._hashRef(Reference(m.b.b1[:].v1[x])) for x in S)
+                ref_data.update(self._hashRef(Reference(m.b.b1[:].v2[x,j])) for x, j in SC)
+                ref_data.update(self._hashRef(Reference(m.b.b1[:].b_s[x].v0)) for x in S)
+                ref_data.update(self._hashRef(Reference(m.b.b1[:].b_s[x1].v1[x2])) for x1, x2 in SS)
+                ref_data.update(self._hashRef(Reference(m.b.b1[:].b_s[x1].v2[x2,j])) for x1,x2,j in SSC)
+                ref_data.update(self._hashRef(Reference(m.b.b2[:,x].v0)) for x in S)
+                ref_data.update(self._hashRef(Reference(m.b.b2[:,x].v1[j])) for x, j in SC)
+                assert len(comps) == len(ref_data)
+                for comp in comps:
+                    self.assertIn(self._hashRef(comp), ref_data)
+            elif len(sets) == 2 and sets[0] is m.time and sets[1] is m.time:
+                ref_data = {
+                    self._hashRef(Reference(m.v_tt)),
+                }
+                ref_data.update(self._hashRef(Reference(m.v_tst[:,x,:])) for x in S)
+                ref_data.update(self._hashRef(Reference(m.b.b2[:,x].v2[:,j])) for x, j in SC)
+                assert len(comps) == len(ref_data)
+                for comp in comps:
+                    self.assertIn(self._hashRef(comp), ref_data)
+            else:
+                raise RuntimeError()
+
+    def _model_2(self):
```

Comment:
Can you provide more descriptive model names (maybe incorporate some of the subsequent comment into the function name; possibly something like `_model2_nD_sets`)

Reply:
Done